presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -518,7 +518,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
                         .forEach(clause -> map(clause.getRight(), clause.getLeft()));
             }
 
-            return new JoinNode(node.getId(), node.getType(), left, right, canonicalCriteria, canonicalFilter, canonicalLeftHashSymbol, canonicalRightHashSymbol);
+            return new JoinNode(node.getId(), node.getType(), left, right, canonicalCriteria, canonicalizeAndDistinct(node.getOutputSymbols()), canonicalFilter, canonicalLeftHashSymbol, canonicalRightHashSymbol);
```

Comment:
OK. I wasn't sure that this was the invariant for `UnaliasSymbolReferences`. It's not obvious since some nodes do not support pruning. I guess the invariant is preserved since mapping canonical symbols causes mapping node to drop one of the duplicates form the output. I wonder if `checkState(no duplicates in output)` would be useful after each visitor